travis_docs_builder.py

```diff
@@ -167,13 +167,48 @@ def setup_GitHub_push(repo):
         'https://{token}@github.com/{repo}.git'.format(token=token.decode('utf-8'), repo=repo)])
     print("Fetching token remote")
     run(['git', 'fetch', 'origin_token'])
+    #create gh-pages empty branch with .nojekyll if it doesn't already exist
+    create_gh_pages()
     print("Checking out gh-pages")
-    # TODO: Create the remote gh-pages if it doesn't exist
     run(['git', 'checkout', '-b', 'gh-pages', '--track', 'origin_token/gh-pages'])
     print("Done")
 
     return True
 
+def gh_pages_exists():
+    """
+    Check if there is a remote gh-pages branch.
+    This isn't completely robust.  If there are multiple remotes and you have a gh-pages
+    branch on the non-default remote, this won't see it.
+    """
+    remote_name = 'origin_token'
+    branch_names = subprocess.check_output(['git', 'branch', '-r']).decode('utf-8').split()
+
+    return '{}/gh-pages'.format(remote_name) in branch_names
+
+def create_gh_pages():
+    """
+    If there is no remote gh-pages branch, create one
+
+    Return True if gh-pages was created, False if not
+    """
+    if not gh_pages_exists():
+        print("Creating gh-pages branch")
+        run(['git', 'checkout', '--orphan', 'gh-pages'])
+        #delete everything in the new ref.  this is non-destructive to existing
+        #refs/branches, etc...
+        run(['git', 'rm', '--cached', '-rf', '.'])
```

Comment:
dammit.  that was silly.  fix incoming

On Thu, Jul 21, 2016 at 3:29 PM, Aaron Meurer <notifications@github.com>
wrote:

> In travis_docs_builder.py
> <https://github.com/gforsyth/travis_docs_builder/pull/13#discussion_r71771259>
> :
>
> > +    branch_names = subprocess.check_output(['git', 'branch', '-r']).decode('utf-8').split()
> > +
> > +    return '{}/gh-pages'.format(remote_name) in branch_names
> > +
> > +def create_gh_pages():
> > +    """
> > +    If there is no remote gh-pages branch, create one
> > +
> > +    Return True if gh-pages was created, False if not
> > +    """
> > +    if not gh_pages_exists():
> > +        print("Creating gh-pages branch")
> > +        run(['git', 'checkout', '--orphan', 'gh-pages'])
> > +        #delete everything in the new ref.  this is non-destructive to existing
> > +        #refs/branches, etc...
> > +        run(['git', 'rm', '--cached', '-rf', '.'])
>
> Oh that's specifically what the --cached option does:
>
>        --cached
>            Use this option to unstage and remove paths only from the index. Working tree files, whether modified or not, will be left alone.
>
> (from git help rm)
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/gforsyth/travis_docs_builder/pull/13/files/c94a76f0ef31f38377a78d9db1ea59ba8a6527c1#r71771259>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbix8qBlpRbjeZeAkd6JtcA0JDsBZ93ks5qX8iwgaJpZM4JP-KE>
> .
>